temporal-sdk/src/main/java/io/temporal/worker/WorkflowTaskDispatchHandle.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker;
+
+import io.temporal.api.workflowservice.v1.PollWorkflowTaskQueueResponse;
+import io.temporal.internal.worker.WorkflowTask;
+import java.io.Closeable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Semaphore;
+import java.util.function.Consumer;
+
+public class WorkflowTaskDispatchHandle implements Closeable {
+  private final CompletableFuture<Void> completed = new CompletableFuture<>();
```

Comment:
Took me a sec to understand what was happening here. I think an `AtomicBoolean` would be clearer

Reply:
Sure, I don't see a big difference with a uncompleted/completed CompletableFuture or an AtomicBoolean for this usage.